File: app/src/main/java/fi/tuska/jalkametri/activity/StatisticsSummaryActivity.java
package fi.tuska.jalkametri.activity;

import android.os.Bundle;
import fi.tuska.jalkametri.R;

/**
 * Shows statistics about your drinking habits. The truth is often harsh;
 * ignorance bliss.
 *
 * @author Tuukka Haapasalo
 */
public class StatisticsSummaryActivity extends AbstractStatisticsActivity {

    public static final String TAG = "StatisticsSummaryActivity";

    public StatisticsSummaryActivity() {
        super();
    }

    /*
     * Functions for preparing the activity to be shown
     * --------------------------------------------
     */

    /*
     * Standard activity functions
     * --------------------------------------------
     */
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.statistics_summary);

        setStatistics(statistics.getGeneralStatistics());
    }

}

File: app/src/main/java/fi/tuska/jalkametri/activity/StatisticsSummaryActivity.kt
package fi.tuska.jalkametri.activity

import android.os.Bundle
import android.widget.TextView
import fi.tuska.jalkametri.R
import fi.tuska.jalkametri.dao.GeneralStatistics
import fi.tuska.jalkametri.util.NumberUtil

class StatisticsSummaryActivity : AbstractStatisticsActivity<StatisticsSummaryActivity.ViewModel>() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStatistics(statisticsDB.generalStatistics)
        setContentView(R.layout.activity_statistics_summary)
    }

    override fun createViewModel() = ViewModel(this)

    class ViewModel(activity: StatisticsSummaryActivity) : AbstractStatisticsActivity.ViewModel(activity) {
        private val firstDay = activity.findViewById(R.id.first_day) as TextView

        override fun updateUI(generalStats: GeneralStatistics) {
            super.updateUI(generalStats)
            generalStats.firstDay.let { date ->
                firstDay.text = if (date != null) dateFormat.print(date) else res.getString(R.string.stats_no_days)
            }
        }
    }

    companion object {
        val TAG = "StatisticsSummaryActivity"
    }

}
